packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -181,125 +211,560 @@ class CupertinoAlertDialog extends StatelessWidget {
-          style: style,
-          child: child,
-          textAlign: TextAlign.center,
-        ),
+// iOS style layout policy for sizing an alert dialog's content section and action
+// button section.
+//
+// The policy is as follows:
+//
+// If all content and buttons fit on screen:
+// The content section and action button section are sized intrinsically and centered
+// vertically on screen.
+//
+// If all content and buttons do not fit on screen, and iOS is NOT in accessibility mode:
+// A minimum height for the action button section is calculated. The action
+// button section will not be rendered shorter than this minimum.  See
+// [_RenderCupertinoDialogActions] for the minimum height calculation.
+//
+// With the minimum action button section calculated, the content section can
+// take up as much space as is available, up to the point that it hits the
+// minimum button height at the bottom.
+//
+// After the content section is laid out, the action button section is allowed
+// to take up any remaining space that was not consumed by the content section.
+//
+// If all content and buttons do not fit on screen, and iOS IS in accessibility mode:
+// The button section is given up to 50% of the available height. Then the content
+// section is given whatever height remains.
+class _RenderCupertinoDialog extends RenderBox {
+  _RenderCupertinoDialog({
+    RenderBox contentSection,
+    RenderBox actionsSection,
+    double dividerThickness = 0.0,
+    bool isInAccessibilityMode = false,
+  }) : _contentSection = contentSection,
+       _actionsSection = actionsSection,
+       _dividerThickness = dividerThickness,
+       _isInAccessibilityMode = isInAccessibilityMode;
+
+  RenderBox get contentSection => _contentSection;
+  RenderBox _contentSection;
+  set contentSection(RenderBox newContentSection) {
+    if (newContentSection != _contentSection) {
+      if (null != _contentSection) {
+        dropChild(_contentSection);
+      }
+      _contentSection = newContentSection;
+      if (null != _contentSection) {
+        adoptChild(_contentSection);
+      }
+    }
+  }
+
+  RenderBox get actionsSection => _actionsSection;
+  RenderBox _actionsSection;
+  set actionsSection(RenderBox newActionsSection) {
+    if (newActionsSection != _actionsSection) {
+      if (null != _actionsSection) {
+        dropChild(_actionsSection);
+      }
+      _actionsSection = newActionsSection;
+      if (null != _actionsSection) {
+        adoptChild(_actionsSection);
+      }
+    }
+  }
+
+  bool get isInAccessibilityMode => _isInAccessibilityMode;
+  bool _isInAccessibilityMode;
+  set isInAccessibilityMode(bool newValue) {
+    if (newValue != _isInAccessibilityMode) {
+      _isInAccessibilityMode = newValue;
+      markNeedsLayout();
+    }
+  }
+
+  double get _dialogWidth => isInAccessibilityMode
+      ? _kAccessibilityCupertinoDialogWidth
+      : _kCupertinoDialogWidth;
+
+  final double _dividerThickness;
+
```

Comment:
I assume this needs setters and calls to markNeedsLayout etc right?

Reply:
No, because like the other reference to divider thickness, it currently is unchangeable.  That's why it's marked final.